migrations/001.do.daily-reward-transfers.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE daily_reward_transfers (
+    day DATE NOT NULL,
+    to_address TEXT NOT NULL,
+    amount NUMERIC NOT NULL,
+    PRIMARY KEY (day, to_address)
+);
+
+CREATE TABLE reward_transfer_last_block (
+    last_block INTEGER NOT NULL
+);
+INSERT INTO reward_transfer_last_block (last_block) VALUES (0);
```

Comment:
what about instead we add `block` to `daily_reward_transfers`, and then query that to get the last block seen?

Reply:
This way we don't run the risk of the two tables getting into an inconsistent state

Reply:
We don't actually want the last block with an Event, we want the last "checked" block. So if it's Friday and spark-observer runs, we don't want it to have to check since Sunday (assuming no Transfers since Sunday) for two reasons:

1. We should have already checked all those blocks
2. We will fail because GLIF only allows last 16h40m. Not a big problem bc I have it set to retry, but still doesn't seem like a clean approach.

And I don't see a great way to incorporate what we actually want into daily_reward_transfers. But if you have some specifics in mind for that, I'm open to it!

Reply:
I'm suggesting to put the last checked block in the above table

> And I don't see a great way to incorporate what we actually want (last "checked" block) into daily_reward_transfers. But if you have some specifics in mind for that, I'm open to it!

Why not put the last checked block in `last_checked_block` in that table?

Reply:
Oh I think I see, so you mean for every row, we have a new field called `last_checked_block`? Then do we query by MAX to get the highest block seen?

I wonder if it will end up being slow to do that MAX query after we have hundreds of thousands of rows? In the current implementation it's just a single value that is updated over time.

Also, it feels a bit unrelated to each row. Like the `last_checked_block` for any given row doesn't really tell you anything about that row, and we lose some semantic consistency of the table.

That said, I see the benefit of avoiding the `daily_reward_transfers` table not losing consistency with the `last_checked_block` value. Let me know which you prefer

Reply:
> Oh I think I see, so you mean for every row, we have a new field called `last_checked_block`? Then do we query by MAX to get the highest block seen?

Yeah that's what I'm thinking 👍 

> I wonder if it will end up being slow to do that MAX query after we have hundreds of thousands of rows? In the current implementation it's just a single value that is updated over time.

With an index on the column I don't see performance being an issue. I would rather start correct with this (if it's one table it's consistent by design) and optimize later if necessary.

> Also, it feels a bit unrelated to each row. Like the `last_checked_block` for any given row doesn't really tell you anything about that row, and we lose some semantic consistency of the table.

It tells us at which state of the blockchain this event was received, which is related to the event.

> That said, I see the benefit of avoiding the `daily_reward_transfers` table not losing consistency with the `last_checked_block` value. Let me know which you prefer

I would prefer to fold it into this table. Maybe @bajtos will disagree next week, and if we decide to go that way after all, I think it won't be too bad to follow up again. To me, here, correctness and simplicity outweigh the optimization.

Reply:
Makes sense! Addressed in https://github.com/filecoin-station/spark-stats/pull/102/commits/3939e8086fa6a0e472515946519f512c76283d07, along with the below suggestions

However, we can't run the dry-run anymore on real events because of the 16h40m max lookback, so we'll have to wait til the next rewards are released (or just merge anyway since it shouldn't break anything else)

Reply:
While I have concerns about the extra storage the proposed design uses, I also see how it gives us more information when troubleshooting and how it can unlock new possibilities.

I am fine with going down the path you chose.

How can we check this design's impact on DB performance & storage?